JenkinsFile

```diff
@@ -0,0 +1,108 @@
+    		-DrepositoryId=nexus \
+    		-Durl="${env.ARTIFACT_STORAGE_DEPLOY_URL}" \
+    		-DgroupId="org.venice.beachfront" \
+    		-DgeneratePom=false \
+    		-Dpackaging=tgz \
+    		-Dmaven.repo.local="${root}/.m2/repository" \
+    		-DartifactId=bfalg-ndwi \
+    		-Dversion=${appvers} \
+    	  """
+        sh "rm ${archiveName}"
+    }}
+stage("Dependency Check") {
+        withCredentials([
+          [$class: 'StringBinding', credentialsId: "${env.THREADFIX_API_KEY}", variable: "THREADFIX_KEY"]
+        ]) {
+		dir("bfalg-ndwi") {
+	            sh """
+        	      /jslave/dependency-check/dependency-check/bin/dependency-check.sh --version
+            	      /jslave/dependency-check/dependency-check/bin/dependency-check.sh --project "bfalg-ndwi" --scan '.' --format "XML" --enableExperimental --disableBundleAudit
+	              /bin/curl -v --insecure -H 'Accept: application/json' -X POST --form file=@dependency-check-report.xml ${env.THREADFIX_URL}/rest/applications/${THREADFIX_ID}/upload?apiKey=${THREADFIX_KEY}
+	    	    """
+		}
+	}
+    }
+    stage("Fortify Scans") {
+        withCredentials([
+          [$class: 'StringBinding', credentialsId: "${env.THREADFIX_API_KEY}", variable: "THREADFIX_KEY"]
+        ]) {
+		dir("bfalg-ndwi") {
+			sh """
+          		  export PATH="$root/miniconda2/bin:$PATH" && source activate bfalg-ndwi
+          		  /opt/hp_fortify_sca/bin/sourceanalyzer -b ${fortifyBuild} -python-path `python -c "import sys;print ':'.join(sys.path)"` {*.py,**/*.py}
+          		  /opt/hp_fortify_sca/bin/sourceanalyzer -b ${fortifyBuild} -scan -python-path `python -c "import sys;print ':'.join(sys.path)"` -Xmx1G -f fortifyResults-${fortifyBuild}.fpr
+	    		  /bin/curl -v --insecure -H 'Accept: application/json' -X POST --form file=@fortifyResults-${fortifyBuild}.fpr ${env.THREADFIX_URL}/rest/applications/${THREADFIX_ID}/upload?apiKey=${THREADFIX_KEY}
+          		  source deactivate
+          		  ls
+        		"""
+       		}
+	}
+    }
+    stage("Sonar") {
+        withCredentials([string(credentialsId: "${env.SONAR_TOKEN}", variable: 'USERNAME')]) {
+            withEnv(["HOME=${root}"]) {
+                sh """export PATH="$root/miniconda2/bin:$PATH" && source activate bfalg-ndwi
+		  ${sonar}/bin/sonar-scanner \
+                    -Dsonar.python.coverage.reportPath=report/coverage/coverage.xml \
+                    -Dsonar.sources=${root}/bfalg-ndwi/bfalg_ndwi \
+                    -Dsonar.host.url=${env.SONAR_URL} \
+                    -Dsonar.projectKey=${projectId} \
+                    -Dsonar.projectName=${projectId} \
+                    -Dsonar.language=py \
+                    -Dsonar.projectBaseDir=${root}/bfalg-ndwi \
+                    -Dsonar.projectVersion=${appvers} \
+                    -Dsonar.login=${USERNAME}
+		  source deactivate
+                """
+            }
+        }
+    }
+    stage("Cleanup") {
```

Comment:
You probably don't even need this stage, because you're doing it before anything happens above:
```
stage("Setup") {
 +        deleteDir()
...
```

Reply:
Agree with removing cleanup, that was a relic of trying to deal with a global condarc